tools/scripts/brokenLinkChecker/brokenLinkChecker.js

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env node
+
+const blc = require('broken-link-checker');
+const https = require('https');
+const url = require('url');
+
+const whitelist = require('./whitelist.json').ignore;
+
+const slackUrl = "https://hooks.slack.com/services/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";
+
+let totalLinkCount = 0;
+let brokenLinkCount = 0;
+let whitedlistedLinkCount = 0;
+
+/* Returns true if the provided string contains one of the whitelisted strings.
+ * Note that it doesn't need to be an exact match; rather a whitelisted string
+ * must simply appear inside what might be a longer provided string.
+ * e.g. if the whitelist contains "testdomain.com/test" then a provided string of
+ * "www.testdomain.com/test/first" will match.
+ */
+function stringContainsWhitelistedString(s) {
+  for (let i = 0; i < whitelist.length; i++) {
+    if (s.includes(whitelist[i])) {
+      return true;
+    }
+  }
```

Comment:
I think you'd have to `return whitelist.some(s.includes.bind(s))` - in which case I feel like the arrow function reads better.